sql/catalyst/src/main/scala/org/apache/spark/sql/QueryExecutionErrors.scala

```diff
@@ -51,18 +58,209 @@ object QueryExecutionErrors {
+    s"Failed to execute user defined function ($funcCls: ($inputTypes) => $outputType)"
+  }
+
+  def failedExecuteUserDefinedFunctionError(
+      funcCls: String, inputTypes: String, outputType: String, e: Exception): Throwable = {
+    new SparkException(udfErrorMessage(funcCls, inputTypes, outputType), e)
+  }
+
+  def divideByZeroError(): Throwable = {
+    new ArithmeticException("divide by zero")
+  }
+
+  def dataTypeUnsupportedByExtractValueError(
+      dataType: DataType, extraction: Expression, child: Expression): Throwable = {
+    val errorMsg = dataType match {
+      case StructType(_) =>
+        s"Field name should be String Literal, but it's $extraction"
+      case other =>
+        s"Can't extract value from $child: need struct type but got ${other.catalogString}"
+    }
+    new AnalysisException(errorMsg)
+  }
+
+  def invalidArrayIndexError(index: Int, numElements: Int): Throwable = {
+    new ArrayIndexOutOfBoundsException(s"Invalid index: $index, numElements: $numElements")
+  }
+
+  def mapKeyNotExistsError(key: Any): Throwable = {
+    new NoSuchElementException(s"Key $key does not exist.")
+  }
+
+  def rowFromCSVParserNotExpectedError(): Throwable = {
+    new IllegalArgumentException("Expected one row from CSV parser.")
+  }
+
+  def inputTypeUnsupportedError(dataType: DataType): Throwable = {
+    new IllegalArgumentException(s"Unsupported input type ${dataType.catalogString}")
+  }
+
+  def invalidFractionOfSecondError(): Throwable = {
+    new DateTimeException("The fraction of sec must be zero. Valid range is [0, 60].")
+  }
+
+  def overflowInSumOfDecimalError(): Throwable = {
+    new ArithmeticException("Overflow in sum of decimals.")
+  }
+
+  def mapSizeExceedArraySizeWhenZipMapError(size: Int): Throwable = {
+    new RuntimeException(s"Unsuccessful try to zip maps with $size " +
+      "unique keys due to exceeding the array size limit " +
+      s"${ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH}.")
+  }
+
+  def copyNullFieldNotAllowedError(): Throwable = {
+    new IllegalStateException("Do not attempt to copy a null field")
+  }
+
+  def literalTypeUnsupportedError(v: Any): Throwable = {
+    new RuntimeException(s"Unsupported literal type ${v.getClass} $v")
+  }
+
+  def noDefaultForDataTypeError(dataType: DataType): Throwable = {
+    new RuntimeException(s"no default for type $dataType")
+  }
+
+  def doGenCodeOfAliasShouldNotBeCalledError(): Throwable = {
+    new IllegalStateException("Alias.doGenCode should not be called.")
+  }
+
+  def orderedOperationUnsupportedByDataTypeError(dataType: DataType): Throwable = {
+    new IllegalArgumentException(s"Type $dataType does not support ordered operations")
+  }
+
+  def indexOfMatchGroupLessThanZeroNotAllowedError(): Throwable = {
+    new IllegalArgumentException("The specified group index cannot be less than zero")
+  }
+
+  def groupIndexExceedMatchGroupCountNotAllowedError(
```

Comment:
regexGroupIndexExceedGroupCountError

Reply:
How about simply saying`exceededRegexGroupIndexError ` instead?

Reply:
In fact, the origin means is regex group index exceeds the regex group count.